### ISSUE TITLE
Bootstrapping deletes installed dependencies

### ISSUE THREAD

**Issue by michaelfoidl** [lerna/lerna#2181]
TL;DR: I have managed to create a scenario where `lerna bootstrap` does not install dependencies specified in a packages' `package.json` file or even deletes them if I had installed them manually before.

I have two packages both depending on electron, in the same version. It is listed in their `devDependencies`:

<details><summary>norris.communication package.json</summary><p>

```json
{
    "name": "norris.communication",
	"main": "./dist/norris.communication.js",
	"types": "./dist/norris.communication.d.ts",
    "version": "0.0.1",
    "private": true,
    "scripts": {
        "checkDependencies": "npm-check",
        "compile": "tsc -p tsconfig.json",
        "clean": "rimraf ./dist",
        "build": "npm run clean && npm run checkDependencies && npm run compile"
    },
    "dependencies": {
        "norris.core": "file:../norris.core",
        "norris.common": "file:../norris.common",
        "rxjs": "~6.5.2"
    },
    "devDependencies": {
        "electron": "~5.0.6",
		"electron-builder": "~21.0.15"
    }
}
```
</p></details>

and 

<details><summary>norris.electron package.json</summary><p>

```json
{
    "name": "norris.electron",
    "version": "0.0.1",
    "private": true,
    "scripts": {
        "checkDependencies": "depcheck",
        "compile": "tsc -p tsconfig.json",
        "clean": "rimraf ./dist",
        "build": "npm run clean && npm run checkDependencies && npm run compile",
        "start": "electron ./dist/main.js",
        "serve": "wait-on http-get://localhost:4200/ && npm run build && electron ./dist/main.js --serve"
    },
    "dependencies": {
        "inversify": "~5.0.1",
        "norris.core": "file:../norris.core",
        "norris.ioc": "file:../norris.ioc",
        "norris.ioc.interfaces": "file:../norris.ioc.interfaces",
        "norris.communication": "file:../norris.communication"
    },
    "devDependencies": {
        "electron": "~5.0.6",
        "electron-builder": "~21.0.15"
    }
}
```
</p></details>

In the `norris.electron` package, everything is installed as expected. In the `norris.communication` package, both `electron` and `electron-builder` are not installed.

## Expected Behavior

Lerna should treat both packages the same and install the necessary dependencies.

## Current Behavior

Somehow, Lerna ignores the dependencies in one of the packages.

## Steps to Reproduce (for bugs)

1. Setup two node packages like to ones above
2. Run `lerna bootstrap --no-hoist *`
3. Check the specific `node_modules` folders if the dependencies were installed correctly

or

1. Setup two node packages like to ones above
2. Run `lerna exec npm install`
3. Check the specific `node_modules` folders if the dependencies were installed correctly.
4. Run `lerna bootstrap --no-hoist *`
5. Check the specific `node_modules` folders again: the dependencies are gone again.

<details><summary>lerna.json</summary><p>

```json
{
  "packages": [
    "packages/*"
  ],
  "version": "0.0.0"
}
```
</p></details>

## Context

I run `lerna bootstrap --no-hoist *` to prevent hoisting, since I encountered issues with angular in other packages: Some where hoisted and could not find their peer dependencies anymore, so I decided to handle hoisting myself b deciding manually which package to put into the top-level `package.json` file. 

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.15.0 |
| `npm --version`  | 6.9.0 |
| `node --version` | 11.14.0 |

| OS | Version |
| --- | --- |
| Windows 10 | 1903 |

**Comment by michaelfoidl** [comment]
I tried to reproduce this behavior with Node 16.15.1, npm 8.11.0 and Lerna 5.0.0 - but everything seems to work fine.
Therefore, I will close this issue.
